x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfig.java

```diff
@@ -117,15 +120,23 @@ public DateHistogramGroupConfig(final String field,
         interval = new DateHistogramInterval(in);
         field = in.readString();
         delay = in.readOptionalWriteable(DateHistogramInterval::new);
-        timeZone = in.readString();
+        // new ZoneId can deal with older DateTimeStrings, no need for version check
+        timeZone = ZoneId.of(in.readString(), ZoneId.SHORT_IDS);
     }
 
     @Override
     public void writeTo(final StreamOutput out) throws IOException {
         interval.writeTo(out);
         out.writeString(field);
         out.writeOptionalWriteable(delay);
-        out.writeString(timeZone);
+        // TODO change version after backport
+        if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
```

Comment:
This should be the version you intend to backport to. Then disable bwc tests in this PR (in the root build.gradle). When doing the backport, remove the single line change that disabled bwc tests. Finally, followup with re-enabling bwc tests on master. This way the followup change has no possibility for accidentally not matching the backcompat logic since it is written correctly from the beginning.